### ISSUE TITLE
3.4 Issue ControlPanel Categories Unable to add modules

### ISSUE THREAD

**Issue by leigh-pointer** [oqtane/oqtane.framework#2673]
It appears that the module categories are not being loaded correctly and this can lead to a situation where once a module is added then it is no longer available.

![image](https://user-images.githubusercontent.com/9447612/224655346-574d58b2-73b6-41b1-b4fa-4bcf0333ccdb.png)

My module is here because it has not been used.  HTML module not available. 

![image](https://user-images.githubusercontent.com/9447612/224656112-718f2454-9624-422a-a78b-b14db9dc4a7a.png)

**Comment by sbwalker** [comment]
please see #2674 - I believe it is related to the problem you are seeing. Basically an enhancement was added in 3.4.0 to prevent information disclosure to unauthenticated users of properties which are only needed server-side. The problem is that ModuleDefinitions are cached server-side - which is essentially a shared instance of data. The logic which was added was only supposed to apply to specific GET method API calls in the SiteController - however because the ModuleDefinitions are shared, it was actually overwriting some of the values in the collection (ie. by reference) which affected other areas of the framework.

**Comment by leigh-pointer** [comment]
Hi @sbwalker
 Just applied the PR and the issue is indeed gone now.  Will there be a patch as the framework is very restricted?

**Comment by sbwalker** [comment]
Yes I was trying to determine the severity of the issue - and after more testing it affects quite a few scenarios (unfortunately). I will publish a 3.4.1 shortly.

**Comment by leigh-pointer** [comment]
#2674 fixes the problem so closing this.
